knowledge_storm/utils.py

```diff
@@ -33,6 +39,198 @@ def load_api_key(toml_file_path):
+
+        try:
+            client = QdrantClient(url=url, api_key=api_key)
+            return VectorStoreManager._check_create_collection(client=client, collection_name=collection_name, model=model)
+        except Exception as e:
+            raise ValueError(f"Error occurs when connecting to the server: {e}")
+
+    def _init_offline_vector_db(vector_store_path: str, collection_name: str, model: HuggingFaceEmbeddings):
+        """
+        Initialize the Qdrant client that is connected to an offline vector store with the given vector store folder path.
+
+        Args:
+            vector_store_path (str): Path to the vector store.
+        """
+        if vector_store_path is None:
+            raise ValueError("Please provide a folder path.")
+
+        try:
+            client = QdrantClient(path=vector_store_path)
+            return VectorStoreManager._check_create_collection(client=client, collection_name=collection_name, model=model)
+        except Exception as e:
+            raise ValueError(f"Error occurs when loading the vector store: {e}")
+    
+    def create_or_update_vector_store(
+
+            vector_db_mode: str,
+            file_path: str,
+            content_column: str,
+            title_column: str = "title",
+            url_column: str = "url",
+            desc_column: str = "description",
+            batch_size: int = 64,
+            chunk_size: int = 500,
+            chunk_overlap: int = 100,
+            collection_name: str = "my_documents",
+            vector_store_path: str = None,
+            url: str =  None,
+            qdrant_api_key: str = None,
+            embedding_model: str = 'BAAI/bge-m3',
+            device: str = "mps",
+    ):
+        """
+        Takes a CSV file where each row is a document and has columns for content, title, url, and description.
+        Then it converts all these documents in the content column to vectors and add them the Qdrant collection.
+
+        Args:
+            vector_store_path (str): Path to the directory where the vector store is stored or will be stored. 
+            vector_db_mode (str): Mode of the Qdrant vector store (offline or online).
+            file_path (str): Path to the CSV file.
+            content_column (str): Name of the column containing the content.
+            title_column (str): Name of the column containing the title. Default is "title".
+            url_column (str): Name of the column containing the URL. Default is "url".
+            desc_column (str): Name of the column containing the description. Default is "description".
+            batch_size (int): Batch size for adding documents to the collection.
+            chunk_size: Size of each chunk if you need to build the vector store from documents.
+            chunk_overlap: Overlap between chunks if you need to build the vector store from documents.
+            collection_name: Name of the Qdrant collection.
```

Comment:
Nit: The `collection_name` seems to be the key parameter here, suggest reordering it as the first parameter for this function and do not provide default value.

Reply:
Next time, make Args order in the comment match the actual arguments order.